### PR TITLE
Avoid calling pad when possible

### LINKED ISSUE #18657: Use the `padding` arg when possible for torch backend
To support `padding="same"`, we implemented a custom padding by calling `torch.nn.functional.pad` for ops including `conv`, `max_pool`, and `average_pool`.

However, it takes more memory and slows down the model.

Torch does support some cases of `padding="same"`.
We should only use this manuall padding method when the native ops in torch doesn't support it.

### PR DESCRIPTION
**Pull request by haifeng-jin**
Resolves #18657

`torch.nn.functional.pad` is slow and memory consuming. We should avoid calling it whenever it is possible.

This PR strictly checks the necessity of calling the function and only call it when necessary.
For all other cases, it just use the `padding` argument of the torch functions like `conv2d`, `max_pool2d`, and `avg_pool2d`.

### CODE REVIEW

File: keras/backend/torch/nn.py [hunk 1]
```diff
@@ -112,20 +111,30 @@ def _compute_padding_length(
...
+    the torch functions. If not compatible, it returns the padded tensor and 0
+    as the padding value.
+
+    Returns:
+        tensor: A padded tensor or the inputs.
+        padding: The padding value, ready to pass to the torch functions.
+    """
     spatial_shape = inputs.shape[2:]
     num_spatial_dims = len(spatial_shape)
     padding = ()
```
Comment (fchollet) [inline keras/backend/torch/nn.py:140]: Looks like it needs to be standardized to a `left, right` tuple.

Comment (haifeng-jin) [reply keras/backend/torch/nn.py:140]: fixed. Thanks!

**Review by fchollet — APPROVED** [review]
LGTM, thank you!
